QuitAll crashes micro

### Description

When you have a file open, performing QuitAll crashes micro with a segfault.

```
Micro encountered an error: runtime.errorString runtime error: invalid memory address or nil pointer dereference
runtime/panic.go:336 (0x48b8d8)
runtime/panic.go:335 (0x48b8a6)
github.com/micro-editor/micro/v2/internal/buffer/buffer.go:691 (0x883dd4)
github.com/micro-editor/micro/v2/internal/buffer/buffer.go:690 (0x883daf)
github.com/micro-editor/micro/v2/internal/buffer/buffer.go:506 (0x8e6f78)
github.com/micro-editor/micro/v2/internal/buffer/buffer.go:504 (0x8e6f33)
github.com/micro-editor/micro/v2/internal/action/actions.go:1969 (0x8cc4b7)
github.com/micro-editor/micro/v2/internal/action/bufpane.go:567 (0x8d3ecd)
github.com/micro-editor/micro/v2/internal/action/tab.go:224 (0x8d22a6)
github.com/micro-editor/micro/v2/internal/action/bufpane.go:183 (0x8d20d9)
github.com/micro-editor/micro/v2/internal/action/bufpane.go:33 (0x8d2071)
github.com/micro-editor/micro/v2/internal/action/bufpane.go:546 (0x8d3c39)
github.com/micro-editor/micro/v2/internal/action/bufpane.go:464 (0x8d3729)
github.com/micro-editor/micro/v2/internal/action/tab.go:337 (0x8e5105)
github.com/micro-editor/micro/v2/internal/action/tab.go:141 (0x8e4245)
github.com/micro-editor/micro/v2/cmd/micro/micro.go:543 (0x90f72a)
github.com/micro-editor/micro/v2/cmd/micro/micro.go:482 (0x90f27b)
internal/runtime/atomic/types.go:194 (0x453c75)
runtime/asm_amd64.s:1771 (0x490f01)

If you can reproduce this error, please report it at https://github.com/micro-editor/micro/issues
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x90e61f]
```

Doing some quick bisecting reveals that the commit ef57bdcf9ba9 from #3997 had introduced this regression.

Adding `QuitAll` to a keybind and running that on any opened file will trigger this behavior.

@JoeKar 



### Environment

- Version: master
- OS: Linux
- Terminal: WezTerm

--- a/internal/buffer/buffer.go
+++ b/internal/buffer/buffer.go
@@ -502,11 +502,14 @@ func NewBuffer(r io.Reader, size int64, path string, btype BufType, cmd Command)
 
 // CloseOpenBuffers removes all open buffers
 func CloseOpenBuffers() {
-	for i, buf := range OpenBuffers {
-		buf.Fini()
-		OpenBuffers[i] = nil
+	// use a copy, since we're gonna remove elements from OpenBuffers
+	// while iterating
+	buffers := make([]*Buffer, len(OpenBuffers))
+	copy(buffers, OpenBuffers)
+
+	for _, buf := range buffers {
+		buf.Close()
 	}
-	OpenBuffers = OpenBuffers[:0]
 }
 
 // Close removes this buffer from the list of open buffers

